src/aiida/orm/nodes/data/code/portable.py

```diff
@@ -177,3 +179,16 @@ def filepath_executable(self, value: str) -> None:
             raise ValueError('The `filepath_executable` should not be absolute.')
 
         self.base.attributes.set(self._KEY_ATTRIBUTE_FILEPATH_EXECUTABLE, value)
+
+    def _prepare_yaml(self, *args, **kwargs):
+        """Export code to a YAML file."""
+        target_path = pathlib.Path().cwd() / f'portablecode-{self.pk}'
+        setattr(self, 'filepath_files', str(target_path))
+        self.base.repository.copy_tree(target=target_path)
+        _LOGGER.warning(f'Repository files for PortableCode <{self.pk}> dumped to folder `{target_path}`.')
+
+        return super()._prepare_yaml(*args, **kwargs)
```

Comment:
I think it would be good to undo the monkeypatch after you are done. It might have some unintended consequences if we leave it there. SO something like
```suggestion
        try:
            setattr(self, 'filepath_files', str(target_path))
            self.base.repository.copy_tree(target=target_path)
            _LOGGER.warning(f'Repository files for PortableCode <{self.pk}> dumped to folder `{target_path}`.')
            result = super()._prepare_yaml(*args, **kwargs)
        finally:
            try:
                delattr(self, 'filepath_files')
            except AttributeError:
                pass
           
        return result
```